### DISCUSSION TITLE
Does Atata work with Angular?

### DISCUSSION THREAD

**Discussion by jerone** [atata-framework/atata#729] [Q&A] [answered]
As the title says; does Atata work with Angular out of the box?

If not, does it work with [Protractor for .NET](https://github.com/bbaia/protractor-net)?

**Comment by YevgeniyShunevych** [comment]
Indeed, as well as Atata works with any sites based on any other web frameworks.

**Reply by jerone** [reply]
As Angular applications are SPA applications, how does Atata know when the page is loaded?

**Reply by YevgeniyShunevych** [reply, marked as answer]
Usually, most cases are handled automatically by built-in Atata auto waits and retries.

But I have just released [Atata v2.6.0](https://atata.io/blog/2023/01/09/atata-2.6.0-released/) with added `WaitForAngular` trigger and method. Check out the "Wait for Angular" section in the mentioned release notes. You can apply `WaitForAngular` trigger globally (it is similar to what Protractor.NET does):

```cs
AtataContext.GlobalConfiguration
    .Attributes.Global.Add(new WaitForAngularAttribute(TriggerEvents.Init | TriggerEvents.AfterClick));
```

**Reply by jerone** [reply]
CoolT When I find the time, I'll investigate if we can implement this in our E2E project.
